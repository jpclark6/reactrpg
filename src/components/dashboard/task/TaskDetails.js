import React from 'react'

const TaskDetails = ({task}) => {
    var classes;
    if (task.completed === true) {
        classes = "card z-depth-0 skill-summary flow-text grey lighten-3"
    } else {
        classes = "card z-depth-0 skill-summary flow-text lighten-3"
    }
    return (
        <div className={classes}>
            <div className="card-content grey-text text-darken-3">
                <span className="card-title">{task.title}</span>
                <p>{task.description}</p>
                <p className="grey-text">{task.category} - difficulty: {task.difficulty}, time: {task.time} </p>
            </div>
        </div>
    )
}

export default TaskDetails;