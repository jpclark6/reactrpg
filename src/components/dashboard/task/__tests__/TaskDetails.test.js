import React from 'react';
import TaskDetails from '../TaskDetails';
import renderer from 'react-test-renderer';

test('TaskDetails are displayed', () => {
    const task = {
        id: '1',
        title: 'Example task',
        description: 'Load other tasks',
        reward: '',
        category: 'adulting',
        difficulty: 3,
        time: 1,
        recurring: 'none',
        completed: false,
        successful: false,
        createdAt: {
            seconds: 1589924745,
            nanoseconds: 4923
        }
    }
    const component = renderer.create(
        <TaskDetails task={task} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});