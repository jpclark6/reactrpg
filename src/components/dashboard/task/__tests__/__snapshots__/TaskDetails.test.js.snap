// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetails are displayed 1`] = `
<div
  className="card z-depth-1 skill-summary flow-text center"
>
  <div
    className="card-content grey-text text-darken-3"
  >
    <span
      className="card-title black-text"
    >
      "
      Example task
      "
    </span>
    <p>
      Load other tasks
    </p>
    <p
      className="grey-text"
    >
      Created 
      <time
        dateTime="2020-05-19T21:45:45.000Z"
        title="5/19/2020, 3:45:45 PM"
      >
        18 hours ago
      </time>
    </p>
    
  </div>
</div>
`;
